### PR TITLE
Added support for validation rules for mixin builders.

### PR DESCRIPTION
**Pull request by rmadsen-ks**
- Added validation rules for mixin builders
- Added test that verifies the funcitonality.
- Added a fix that caused only basic type info to be shown in the MenuAnnotations.

### CODE REVIEW

File: Engine.UnitTests/EmbeddedPropertyTests.cs [hunk 1]
```diff
@@ -146,12 +146,11 @@ public void OnPostRun(TestStepPostRunEventArgs step)
             }
         }
         
-        public class RepeatMixin : ITestStepPostRunMixin, ITestStepPreRunMixin
+        public class RepeatMixin2 : ITestStepPostRunMixin, ITestStepPreRunMixin
```
Comment (alnlarsen) [inline Engine.UnitTests/EmbeddedPropertyTests.cs:149]: Why was this renamed?

**Review by alnlarsen — COMMENTED** [review]
This all looks good but I'm wondering why the test mixin was renamed

File: Engine.UnitTests/EmbeddedPropertyTests.cs (hunk 1 above)
Comment (rmadsen-ks) [reply Engine.UnitTests/EmbeddedPropertyTests.cs:149]: Just to avoid clashing with a mixin defined by Basic Mixins. Its just a unit test, so it should not cause a problem.

**Review by alnlarsen — APPROVED** [review]
Looks good :)
